field.go

```diff
@@ -68,6 +87,35 @@ func (f *Field) AvailableMoves() []*Move {
 	return moves
 }
 
+func (f *Field) Hash() uint32 {
+	f.zhash = 0
+	for col := range f.field {
+		for row := range f.field[col] {
+			var point int = 0
+			if f.field[col][row] == f.MyId {
+				point = 1
+			} else if f.field[col][row] == f.OpponentId {
+				point = 2
+			}
```

Comment:
Lines 94-99 can be abstracted into their own function to avoid copying them below